File: navigation-compiler/src/main/java/com/arasthel/navigation/annotations/RegisterScreen.java
package com.arasthel.navigation.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import com.arasthel.navigation.screen.Screen;
import com.arasthel.navigation.screen.ScreenResult;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.CLASS)
public @interface RegisterScreen {
	Class<? extends Screen> value();

	Class<? extends ScreenResult> screenResult() default ScreenResult.class;
}

File: annotation-processor/src/main/java/com/arasthel/navigation/annotations/RegisterScreen.kt
package com.arasthel.navigation.annotations

import com.arasthel.navigation.screen.Screen
import kotlin.reflect.KClass

/**
 * Can be used to register activities, fragments and dialog fragments with the help of an annotation processor.
 */
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class RegisterScreen(
    val value: KClass<out Screen>,
)